### PR TITLE
feat(config): support retrieving configuration fields using glob patterns

### PR DESCRIPTION
**Pull request by sryanyuan**
It can be difficult to remember the full configuration field, so providing a keyword to retrieve the configuration value, as is done in Redis, can be more convenient.

### CODE REVIEW

File: src/vendor/globmatch.cc [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "globmatch.h"
+
+// NOLINTBEGIN
+/* Glob-style pattern matching. */
+int stringmatchlen(const char *pattern, int patternLen, const char *string, int stringLen, int nocase) {
```
Comment (PragmaTwice) [inline src/vendor/globmatch.cc]: Seems we already have one called `StringMatch` in https://github.com/apache/kvrocks/blob/unstable/src/common/string_util.cc .

Comment (sryanyuan) [reply src/vendor/globmatch.cc]: ok, i'll use this one

Comment (sryanyuan) [reply src/vendor/globmatch.cc]: fixed

File: src/config/config.cc [hunk 2]
```diff
@@ -867,7 +868,7 @@ Status Config::Load(const CLIOptions &opts) {
 void Config::Get(const std::string &key, std::vector<std::string> *values) const {
   values->clear();
   for (const auto &iter : fields_) {
-    if (key == "*" || util::ToLower(key) == iter.first) {
+    if (util::StringMatchLen(key.c_str(), key.size(), iter.first.c_str(), iter.first.size(), 1)) {
```
Comment (PragmaTwice) [inline src/config/config.cc]:
```suggestion
    if (util::StringMatch(key, iter.first, 1)) {
```

Comment (sryanyuan) [reply src/config/config.cc]: ok

Comment (sryanyuan) [reply src/config/config.cc]: Done

**Review by PragmaTwice — APPROVED** [review]
Looks good to me.

**Review by git-hulk — APPROVED** [review]

**Pull request merged**